providers/go-feature-flag/src/main/java/dev/openfeature/contrib/providers/gofeatureflag/GoFeatureFlagProvider.java

```diff
@@ -262,15 +260,10 @@ private <T> ProviderEvaluation<T> getEvaluation(
         GoFeatureFlagUser user = GoFeatureFlagUser.fromEvaluationContext(evaluationContext);
         try {
             if (!ProviderState.READY.equals(state)) {
-                ErrorCode errorCode = ErrorCode.PROVIDER_NOT_READY;
-                if (ProviderState.ERROR.equals(state)) {
-                    errorCode = ErrorCode.GENERAL;
+                if (ProviderState.NOT_READY.equals(state)) {
+                    throw new ProviderNotReadyError("provider not initialized yet");
                 }
-                return ProviderEvaluation.<T>builder()
-                        .errorCode(errorCode)
-                        .reason(errorCode.name())
-                        .value(defaultValue)
-                        .build();
+                throw new GeneralError("unknown error, provider state: " + state);
```

Comment:
I think @thomaspoignant needs to approve the usage of evaluation response with error vs throwing an exception

Reply:
see [this](https://github.com/open-feature/java-sdk/pull/561#discussion_r1295058795) comment by @toddbaert :

> throwing like this ends in the same result, but it's a bit easier to trace, and also simpler to write. This is how we generally recommend it's done. There is a small performance cost in serialization of the exception, but I think it's worth it.

When is ProviderNotReadyError thrown here, it should be handled by the SDK framework, ErrorCode.PROVIDER_NOT_READY and default value should be returned when evaluated via the client.

I just added a [test](https://github.com/open-feature/java-sdk-contrib/pull/403/files#diff-3d0ec86c96fe8d2f9224111448f23a4ccad41ee847a904fb50fd15921623b7faR187) step to enforce it.

As it is not always clear and trivial that throwing ProviderNotReadyError is handled by the framework, if you have any suggestion on how to make it clear in documentation/spec/provider guidelines, you are welcome to share your thoughts.

Reply:
> As it is not always clear and trivial that throwing ProviderNotReadyError is handled by the framework..

Yeah, that's true. It's not trivial. Also, we have an error logging for exception [^1] which makes this slightly different.

I will wait for @thomaspoignant approval on this :) Otherwise, change looks good 

[^1]: https://github.com/open-feature/java-sdk/blob/v1.5.0/src/main/java/dev/openfeature/sdk/OpenFeatureClient.java#L140